### PR TITLE
Feature: create simple routing api and replace existing routes

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This introduces a basic routing api and replaces existing Next Gen form routes with the `givewp-route` param.

It's purpose is to clean up how we're handling our custom _query param_ based routing so it's easier to create and understand them.

When declaring a route, the uri will turn into `/?givewp-route=$uri`. 

Example:

**Get route url**

```
Route::url('donate') // http://{givewp.com}/?givewp-route=donate
```

**Simple GET route using invokable class**

```
Route::get('donation-form-view', InvokableDonationFormViewController::class);

This will call give(InvokableDonationFormViewController::class)->__invoke(array $request);
```

**GET Route using callable**

```
Route::get('donation-form-view', static function (array $request) {
    $routeData = DonationFormViewRouteData::fromRequest($request);

    return give(DonationFormViewController::class)->show($routeData);
});

```

**POST Route using callable**

```
Route::post('donation-form-view-preview', static function (array $request) {
  $routeData = DonationFormPreviewRouteData::fromRequest($request);

  return give(DonationFormViewController::class)->preview($routeData);
});
```

## Affects

This does technically touch all the working Next Gen routes for viewing, previewing, and donating.

## Visuals

N/A

## Testing Instructions

Just make sure viewing, previewing, and donating a next gen form still works properly.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@kjohnson this is ready for review.  It's not meant to be full-featured, just useful enough for our use cases and we can expand later if need be.  I'll be using this for the new receipt routing.

File: src/NextGen/DonationForm/ServiceProvider.php [hunk 1]
```diff
@@ -29,11 +32,39 @@ public function boot()
...
+            $routeData = DonationFormViewRouteData::fromRequest($request);
+
+            return give(DonationFormViewController::class)->show($routeData);
+        });
+
+
+        /**
+         * @unreleased
+         */
+        Route::post('donation-form-view-preview', static function (array $request) {
```
Comment (kjohnson) [inline src/NextGen/DonationForm/ServiceProvider.php:64]: These inline routes are beautiful.

**Review by kjohnson — APPROVED** [review]
This looks great. Nice to see a `Route::get()` syntax that is clean and easy to reason about.

File: src/NextGen/DonationForm/ServiceProvider.php (hunk 1 above)
Comment (jonwaldstein) [reply src/NextGen/DonationForm/ServiceProvider.php:64]: I know right! It's a start lol

**Pull request merged**